bn-apps/memberships-management/membership-service/src/test/kotlin/com/r3/businessnetworks/membership/flows/SelfIssueMembershipFlowTest.kt

```diff
@@ -1,51 +1,51 @@
 package com.r3.businessnetworks.membership.flows
 
-import com.r3.businessnetworks.membership.flows.bno.RequestMembershipFlowResponder
+import com.r3.businessnetworks.membership.flows.bno.SelfIssueMembershipFlow
 import com.r3.businessnetworks.membership.flows.bno.service.DatabaseService
-import com.r3.businessnetworks.membership.flows.member.service.MemberConfigurationService
 import com.r3.businessnetworks.membership.states.MembershipContract
 import com.r3.businessnetworks.membership.states.MembershipState
-import com.r3.businessnetworks.membership.testextensions.RequestMembershipFlowResponderWithMetadataVerification
 import net.corda.core.flows.FlowException
+import net.corda.core.utilities.getOrThrow
 import org.junit.Test
-import kotlin.test.assertEquals
-import kotlin.test.assertTrue
 import kotlin.test.fail
 
-class SelfIssueMembershipFlowTest : AbstractFlowTest(numberOfBusinessNetworks = 2,
-        numberOfParticipants = 2,
-        participantRespondingFlows = listOf(RequestMembershipFlowResponder::class.java)) {
+class SelfIssueMembershipFlowTest : AbstractFlowTest(
+        numberOfBusinessNetworks = 2,
+        numberOfParticipants = 4,
+        participantRespondingFlows = listOf(NotificationsCounterFlow::class.java)) {
 
     @Test
     fun `self issue happy path`() {
         val bnoNode = bnoNodes.first()
-
+        //membership state before activation
         val stx = runSelfIssueMembershipFlow(bnoNode)
-        
-        assert(stx.notary!!.name == notaryName)
```

Comment:
Plain `assert`s provide unhelpful error messages. I believe this Gradle project has dependencies to two proper testing frameworks: kotlintest and hamkrest (my personal favourite).

Using a testing frameworks will give you nice, understandable error output like this:

_expected: a value that is equal to "com.r3.businessnetworks.membership.states.MembershipContract"
but was: "com.example.bogus"_

whereas plain asserts give you something like that:

_java.lang.AssertionError at SelfIssueMembershipFlowTest.kt:14_

Reply:
I see, I'll add helpful custom messages

Reply:
No need to add messages. Using a proper matcher will generate these for you. Have a look at https://github.com/npryce/hamkrest/blob/aedce9d62adaa1d13226d3037701e2d74c078bd6/src/commonTest/kotlin/com/natpryce/hamkrest/describe_tests.kt for examples

Reply:
I.e. turningg `assert(MembershipContract.CONTRACT_NAME == outputTxState.contract)` into `assertThat(outputTxState.contract, equalTo(MembershipContract.CONTRACT_NAME))` and `assert(command.value is MembershipContract.Commands.Activate)` into `assertThat(command.value, isA<Activate>())` will give you high quality failure messages.